From bac3a4e435102d2257d0781ef9521d9f135aca9c Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Wed, 23 Oct 2019 10:32:47 +0000
Subject: [PATCH] arm64: dts: renesas: eView coexecution

remove peripheral drivers used by CR7 bare-metal application:
   - CSI40 & CSI41
   - DU
   - FCPV
   - I2C (channels 0 & 1)
   - IMR (all channels)
   - LVDS
   - VIN (all channels)
   - VSPD
---
 arch/arm64/boot/dts/renesas/r8a77980.dtsi | 342 ------------------------------
 1 file changed, 342 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77980.dtsi b/arch/arm64/boot/dts/renesas/r8a77980.dtsi
index c5859629..991f8a5 100644
--- a/arch/arm64/boot/dts/renesas/r8a77980.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a77980.dtsi
@@ -17,8 +17,6 @@
 	#size-cells = <2>;
 
 	aliases {
-		i2c0 = &i2c0;
-		i2c1 = &i2c1;
 		i2c2 = &i2c2;
 		i2c3 = &i2c3;
 		i2c4 = &i2c4;
@@ -441,40 +439,6 @@
 			status = "disabled";
 		};
 
-		i2c0: i2c@e6500000 {
-			compatible = "renesas,i2c-r8a77980",
-				     "renesas,rcar-gen3-i2c";
-			reg = <0 0xe6500000 0 0x40>;
-			interrupts = <GIC_SPI 287 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 931>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 931>;
-			dmas = <&dmac1 0x91>, <&dmac1 0x90>,
-			       <&dmac2 0x91>, <&dmac2 0x90>;
-			dma-names = "tx", "rx", "tx", "rx";
-			i2c-scl-internal-delay-ns = <6>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-			status = "disabled";
-		};
-
-		i2c1: i2c@e6508000 {
-			compatible = "renesas,i2c-r8a77980",
-				     "renesas,rcar-gen3-i2c";
-			reg = <0 0xe6508000 0 0x40>;
-			interrupts = <GIC_SPI 288 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 930>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 930>;
-			dmas = <&dmac1 0x93>, <&dmac1 0x92>,
-			       <&dmac2 0x93>, <&dmac2 0x92>;
-			dma-names = "tx", "rx", "tx", "rx";
-			i2c-scl-internal-delay-ns = <6>;
-			#address-cells = <1>;
-			#size-cells = <0>;
-			status = "disabled";
-		};
-
 		i2c2: i2c@e6510000 {
 			compatible = "renesas,i2c-r8a77980",
 				     "renesas,rcar-gen3-i2c";
@@ -888,174 +852,6 @@
 			status = "disabled";
 		};
 
-		vin0: video@e6ef0000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef0000 0 0x1000>;
-			interrupts = <GIC_SPI 188 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 811>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 811>;
-			renesas,id = <0>;
-			status = "disabled";
-		};
-
-		vin1: video@e6ef1000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef1000 0 0x1000>;
-			interrupts = <GIC_SPI 189 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 810>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			status = "disabled";
-			renesas,id = <1>;
-			resets = <&cpg 810>;
-		};
-
-		vin2: video@e6ef2000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef2000 0 0x1000>;
-			interrupts = <GIC_SPI 190 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 809>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 809>;
-			renesas,id = <2>;
-			status = "disabled";
-		};
-
-		vin3: video@e6ef3000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef3000 0 0x1000>;
-			interrupts = <GIC_SPI 191 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 808>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 808>;
-			renesas,id = <3>;
-			status = "disabled";
-		};
-
-		vin4: video@e6ef4000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef4000 0 0x1000>;
-			interrupts = <GIC_SPI 174 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 807>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 807>;
-			renesas,id = <4>;
-			status = "disabled";
-		};
-
-		vin5: video@e6ef5000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef5000 0 0x1000>;
-			interrupts = <GIC_SPI 175 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 806>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 806>;
-			renesas,id = <5>;
-			status = "disabled";
-		};
-
-		vin6: video@e6ef6000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef6000 0 0x1000>;
-			interrupts = <GIC_SPI 176 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 805>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 805>;
-			renesas,id = <6>;
-			status = "disabled";
-		};
-
-		vin7: video@e6ef7000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef7000 0 0x1000>;
-			interrupts = <GIC_SPI 171 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 804>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 804>;
-			renesas,id = <7>;
-			status = "disabled";
-		};
-
-		vin8: video@e6ef8000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef8000 0 0x1000>;
-			interrupts = <GIC_SPI 268 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 628>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 628>;
-			status = "disabled";
-		};
-
-		vin9: video@e6ef9000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6ef9000 0 0x1000>;
-			interrupts = <GIC_SPI 269 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 627>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 627>;
-			status = "disabled";
-		};
-
-		vin10: video@e6efa000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6efa000 0 0x1000>;
-			interrupts = <GIC_SPI 289 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 625>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 625>;
-			status = "disabled";
-		};
-
-		vin11: video@e6efb000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6efb000 0 0x1000>;
-			interrupts = <GIC_SPI 296 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 618>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 618>;
-			status = "disabled";
-		};
-
-		vin12: video@e6efc000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6efc000 0 0x1000>;
-			interrupts = <GIC_SPI 298 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 612>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 612>;
-			status = "disabled";
-		};
-
-		vin13: video@e6efd000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6efd000 0 0x1000>;
-			interrupts = <GIC_SPI 299 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 608>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 608>;
-			status = "disabled";
-		};
-
-		vin14: video@e6efe000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6efe000 0 0x1000>;
-			interrupts = <GIC_SPI 301 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 605>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 605>;
-			status = "disabled";
-		};
-
-		vin15: video@e6eff000 {
-			compatible = "renesas,vin-r8a77980";
-			reg = <0 0xe6eff000 0 0x1000>;
-			interrupts = <GIC_SPI 302 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 604>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 604>;
-			status = "disabled";
-		};
-
 		imp_distributer: impdes0 {
 			compatible = "renesas,impx5+-distributer";
 			reg = <0 0xffa00000 0 0x10000>;
@@ -1239,72 +1035,6 @@
 			dma-coherent;
 		};
 
-		imrlx4_ch0: imr0@fe860000 {
-			compatible = "renesas,r8a77980-imr-lx4",
-				     "renesas,imr-lx4";
-			reg = <0 0xfe860000 0 0x10000>;
-			interrupts = <GIC_SPI 192 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 823>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 823>;
-			alloc-dev = <&imr_v4l2_alloc>;
-		};
-
-		imrlx4_ch1: imr1@fe870000 {
-			compatible = "renesas,r8a77980-imr-lx4",
-				     "renesas,imr-lx4";
-			reg = <0 0xfe870000 0 0x10000>;
-			interrupts = <GIC_SPI 193 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 822>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 822>;
-			alloc-dev = <&imr_v4l2_alloc>;
-		};
-
-		imrlx4_ch2: imr2@fe880000 {
-			compatible = "renesas,r8a77980-imr-lx4",
-				     "renesas,imr-lx4";
-			reg = <0 0xfe880000 0 0x10000>;
-			interrupts = <GIC_SPI 194 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 821>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 821>;
-			alloc-dev = <&imr_v4l2_alloc>;
-		};
-
-		imrlx4_ch3: imr3@fe890000 {
-			compatible = "renesas,r8a77980-imr-lx4",
-				     "renesas,imr-lx4";
-			reg = <0 0xfe890000 0 0x10000>;
-			interrupts = <GIC_SPI 195 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 820>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 820>;
-			alloc-dev = <&imr_v4l2_alloc>;
-		};
-
-		imrlx4_ch4: imr4@fe8a0000 {
-			compatible = "renesas,r8a77980-imr-lx4",
-				     "renesas,imr-lx4";
-			reg = <0 0xfe8a0000 0 0x10000>;
-			interrupts = <GIC_SPI 254 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 707>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			rse;
-			alloc-dev = <&imr_v4l2_alloc>;
-		};
-
-		imrlx4_ch5: imr5@fe8b0000 {
-			compatible = "renesas,r8a77980-imr-lx4",
-				     "renesas,imr-lx4";
-			reg = <0 0xfe8b0000 0 0x10000>;
-			interrupts = <GIC_SPI 255 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 706>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			rse;
-			alloc-dev = <&imr_v4l2_alloc>;
-		};
-
 		vip_disp_status {
 			compatible = "generic-uio";
 			reg = <0 0xe7a00000 0 0x10000>;
@@ -1752,78 +1482,6 @@
 			status = "disabled";
 		};
 
-		vspd0: vsp@fea20000 {
-			compatible = "renesas,vsp2";
-			reg = <0 0xfea20000 0 0x5000>;
-			interrupts = <GIC_SPI 169 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 623>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 623>;
-			renesas,fcp = <&fcpvd0>;
-		};
-
-		fcpvd0: fcp@fea27000 {
-			compatible = "renesas,fcpv";
-			reg = <0 0xfea27000 0 0x200>;
-			clocks = <&cpg CPG_MOD 603>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 603>;
-		};
-
-		csi40: csi2@feaa0000 {
-			compatible = "renesas,r8a77980-csi2";
-			reg = <0 0xfeaa0000 0 0x10000>;
-			interrupts = <GIC_SPI 246 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 716>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 716>;
-			status = "disabled";
-		};
-
-		csi41: csi2@feab0000 {
-			compatible = "renesas,r8a77980-csi2";
-			reg = <0 0xfeab0000 0 0x10000>;
-			interrupts = <GIC_SPI 241 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 715>;
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 715>;
-			status = "disabled";
-		};
-
-		du: display@feb00000 {
-			compatible = "renesas,du-r8a77980";
-			reg = <0 0xfeb00000 0 0x80000>,
-			      <0 0xfeb90000 0 0x14>;
-			reg-names = "du.0", "lvds.0";
-			interrupts = <GIC_SPI 256 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&cpg CPG_MOD 724>,
-				 <&cpg CPG_MOD 727>;
-			clock-names = "du.0", "lvds.0";
-			power-domains = <&sysc R8A77980_PD_ALWAYS_ON>;
-			resets = <&cpg 724>,
-				 <&cpg 727>;
-			reset-names = "du.0", "lvds.0";
-			vsps = <&vspd0>;
-			status = "disabled";
-
-			ports {
-				#address-cells = <1>;
-				#size-cells = <0>;
-
-				port@0 {
-					reg = <0>;
-					du_out_rgb: endpoint {
-					};
-				};
-
-				port@1 {
-					reg = <1>;
-					du_out_lvds0: endpoint {
-					};
-				};
-			};
-		};
-
 		prr: chipid@fff00044 {
 			compatible = "renesas,prr";
 			reg = <0 0xfff00044 0 4>;
-- 
2.7.4

